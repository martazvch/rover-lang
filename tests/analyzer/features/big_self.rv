struct Foo {
    x, y = 0

    fn init() -> Self {
        return Self{} 
    }
}

struct Bar {
    a: []bool

    fn init() -> Self {
        return Self {a=[true]}
    }
}

fn main() {
    struct Baz {
        flag = true
        
        fn init() -> Self {
            return Self {flag=false}
        }
    }
}
