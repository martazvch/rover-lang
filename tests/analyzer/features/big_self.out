[Structure declaration Foo]
    - default values
    [Int 0]
    [Int 0]

    [Function declaration init, returns]
        [Return]
            [Structure literal]
                - structure
                [Load symbol 0]
                - args
                [Default field 0]
                [Default field 1]

[Structure declaration Bar]
    [Function declaration init, returns]
        [Return]
            [Structure literal]
                - structure
                [Load symbol 2]
                - args
                [Array]
                    [Bool true]

[Function declaration main]
    [Structure declaration Baz]
        - default values
        [Bool true]

        [Function declaration init, returns]
            [Return]
                [Structure literal]
                    - structure
                    [Load symbol 5]
                    - args
                    [Bool false]
